programs/mobile-entity-manager/src/instructions/initialize_incentive_program_v0.rs

```diff
@@ -0,0 +1,175 @@
+    bump,
+    space = 60 + IncentiveEscrowProgramV0::INIT_SPACE,
+  )]
+  pub incentive_escrow_program: Box<Account<'info, IncentiveEscrowProgramV0>>,
+  #[account(
+      mut,
+      seeds = [merkle_tree.key().as_ref()],
+      seeds::program = bubblegum_program.key(),
+      bump,
+  )]
+  pub tree_authority: Box<Account<'info, TreeConfig>>,
+  /// CHECK: Used in cpi
+  pub recipient: AccountInfo<'info>,
+  /// CHECK: Used in cpi
+  #[account(mut)]
+  pub merkle_tree: AccountInfo<'info>,
+  #[account(
+    seeds = ["collection_cpi".as_bytes()],
+    seeds::program = bubblegum_program.key(),
+    bump,
+  )]
+  /// CHECK: Used in cpi
+  pub bubblegum_signer: UncheckedAccount<'info>,
+
+  /// CHECK: Verified by constraint
+  #[account(address = mpl_token_metadata::ID)]
+  pub token_metadata_program: AccountInfo<'info>,
+  pub log_wrapper: Program<'info, Noop>,
+  pub bubblegum_program: Program<'info, Bubblegum>,
+  pub compression_program: Program<'info, SplAccountCompression>,
+  pub system_program: Program<'info, System>,
+  pub helium_entity_manager_program: Program<'info, HeliumEntityManager>,
+}
+
+pub fn handler(
+  ctx: Context<InitializeIncentiveProgramV0>,
+  args: InitializeIncentiveProgramArgsV0,
+) -> Result<()> {
+  let seeds: &[&[&[u8]]] = &[carrier_seeds!(ctx.accounts.carrier)];
+
+  issue_program_entity_v0(
+    CpiContext::new_with_signer(
+      ctx.accounts.helium_entity_manager_program.to_account_info(),
+      IssueProgramEntityV0 {
+        payer: ctx.accounts.payer.to_account_info(),
+        program_approver: ctx.accounts.carrier.to_account_info(),
+        program_approval: ctx.accounts.program_approval.to_account_info(),
+        collection_authority: ctx.accounts.carrier.to_account_info(),
+        collection: ctx.accounts.collection.to_account_info(),
+        collection_metadata: ctx.accounts.collection_metadata.to_account_info(),
+        collection_master_edition: ctx.accounts.collection_master_edition.to_account_info(),
+        entity_creator: ctx.accounts.entity_creator.to_account_info(),
+        dao: ctx.accounts.dao.to_account_info(),
+        key_to_asset: ctx.accounts.key_to_asset.to_account_info(),
+        tree_authority: ctx.accounts.tree_authority.to_account_info(),
+        recipient: ctx.accounts.recipient.to_account_info(),
+        merkle_tree: ctx.accounts.merkle_tree.to_account_info(),
+        bubblegum_signer: ctx.accounts.bubblegum_signer.to_account_info(),
+        token_metadata_program: ctx.accounts.token_metadata_program.to_account_info(),
+        log_wrapper: ctx.accounts.log_wrapper.to_account_info(),
+        bubblegum_program: ctx.accounts.bubblegum_program.to_account_info(),
+        compression_program: ctx.accounts.compression_program.to_account_info(),
+        system_program: ctx.accounts.system_program.to_account_info(),
+      },
+      seeds,
+    ),
+    IssueProgramEntityArgsV0 {
+      entity_key: args.name.as_bytes().to_vec(),
+      name: args.name,
```

Comment:
I know IssueProgramEntity has checks on the length of this. We cool with just doing the pass through or should we duplicate the checks here?